pandas/io/formats/format.py

```diff
@@ -366,10 +366,10 @@ def _get_formatted_index(self) -> tuple[list[str], bool]:
 
         if isinstance(index, MultiIndex):
             have_header = any(name for name in index.names)
-            fmt_index = index.format(names=True)
+            fmt_index = index.format(names=True, na_rep=None)
```

Comment:
i think we need to correctly format floats with `NaN` in the indexes itself (if nothing is passed) to have `NaN` (rather than `nan`). see if that works.

Reply:
> i think we need to correctly format floats with `NaN` in the indexes itself (if nothing is passed) to have `NaN` (rather than `nan`). see if that works.

@jreback  True, it seems like a better solution now. 

There are some test failures now, but it looks like not due to this change (maybe docs - but I don't see what went wrong there).

Reply:
excellent!  yeah let me see what's up with the docs